docs/developer-guides/transactions/txscript/opcodes.md

```diff
@@ -0,0 +1,238 @@
+| `OP_AND`    | 132    | 0x84  | Boolean and between each bit in the inputs. |
+| `OP_OR`     | 133    | 0x85  | Boolean or between each bit in the inputs. |
+| `OP_XOR`    | 134    | 0x86  | Boolean exclusive or between each bit in the inputs. |
+
+## Bytewise Comparison
+
+These opcodes compare the top two items on the stack as raw bytes.
+
+| Word             | Opcode | Hex   | Description |
+|------------------|--------|-------|-------------|
+| `OP_EQUAL`       | 135    | 0x87  | Returns 1 if the inputs are exactly equal, 0 otherwise. |
+| `OP_EQUALVERIFY` | 136    | 0x88  | Same as OP_EQUAL, but runs OP_VERIFY afterward. |
+
+## Bitwise Rotation
+
+These opcodes enable circular shifting of bits.
+They treat the top two items on the stack as 32-bit integers, the first
+respresents the number of bits to rotate to the left or right, and the second is
+the value to be rotated.
+
+| Word       | Opcode | Hex   | Description |
+|------------|--------|-------|-------------|
+| `OP_ROTR`  | 137    | 0x89  | Shift bits right, sign is not preserved. |
+| `OP_ROTL`  | 138    | 0x8a  | Shift bits left, sign is not preserved. |
+
+## Arithmetic
+
+These opcodes implement common arithmetic operations to be used on integer
+values.
+
+Arithmetic inputs are limited to signed 32-bit integers, but may overflow their
+output.
+For example, it is possible for `OP_ADD` to have 2^31 - 1 for its two operands
+resulting 2^32 - 2, which overflows, but is still pushed to the stack as the
+result of the addition.
+
+| Word                    | Opcode | Hex   | Description |
+|-------------------------|--------|-------|-------------|
+| `OP_1ADD`               | 139    | 0x8b  | 1 is added to the input. |
+| `OP_1SUB`               | 140    | 0x8c  | 1 is subtracted from the input. |
+| `OP_2MUL`               | 141    | 0x8d  | The input is multiplied by 2. |
+| `OP_2DIV`               | 142    | 0x8e  | The input is divided by 2. |
+| `OP_NEGATE`             | 143    | 0x8f  | The sign of the input is flipped. |
+| `OP_ABS`                | 144    | 0x90  | The input is made positive. |
+| `OP_NOT`                | 145    | 0x91  | If the input is 0 or 1, it is flipped. Otherwise the output will be 0. |
+| `OP_0NOTEQUAL`          | 146    | 0x92  | Returns 0 if the input is 0. 1 otherwise. |
+| `OP_ADD`                | 147    | 0x93  | Performs addition on the inputs. |
+| `OP_SUB`                | 148    | 0x94  | Performs subtraction on the inputs. |
+| `OP_MUL`                | 149    | 0x95  | Performs multiplication on the inputs. |
+| `OP_DIV`                | 150    | 0x96  | Performs division on the inputs. |
+| `OP_MOD`                | 151    | 0x97  | Performs modulo on the inputs. |
+| `OP_LSHIFT`             | 152    | 0x98  | Shifts bits left. Zeros are shifted in on the right. Does not preserve sign. |
+| `OP_RSHIFT`             | 153    | 0x99  | Shifts bits right. Zeros are shifted in on the left. Does not preserve sign. |
```

Comment:
This was correct as it was.  `RSHIFT` **does** preserve the sign, but `LSHIFT` does **NOT**.

e.g. `0x80000001` (aka -2147483647) shifted **right** twice is `0xE0000000` (aka -536870912), but e.g. `0x80000001` (aka -2147483647) shifted **left** twice is `0x00000004` (aka 4).